neo/io/basefromrawio/proxyobjects.py

```diff
@@ -0,0 +1,515 @@
+
+from neo.core.baseneo import BaseNeo
+
+
+from neo.core import (AnalogSignal,
+                      Epoch, Event, SpikeTrain)
+
+
+class BaseProxy(BaseNeo):
+    def __init__(self, **kargs):
+        # this for py27 str vs py3 str in neo attributes ompatibility
+        kargs = check_annotations(kargs)
+        if 'file_origin' not in kargs:
+            kargs['file_origin'] = self._rawio.source_name()
+        BaseNeo.__init__(self, **kargs)
+
+
+class AnalogSignalProxy(BaseProxy):
+    '''
+    This object mimic AnalogSignal except that it does not
+    have the signals array itself. All attributes and annotations are here.
+
+    The goal is to postpone the loading of data into memory
+    when reading a file with the new lazy load system based
+    on neo.rawio.
+
+    This object must not be constructed directly but is given
+    neo.io when lazy=True instead of a true AnalogSignal.
+
+    The AnalogSignalProxy is able to load:
+      * only a slice of time
+      * only a subset of channels
+      * have an internal raw magnitude identic to the file (int16) with
+        a pq.CompoundUnit().
+
+    Usage:
+    >>> proxy_anasig = AnalogSignalProxy(rawio=self.reader,
+                                                                global_channel_indexes=None,
+                                                                block_index=0,
+                                                                seg_index=0)
+    >>> anasig = proxy_anasig.load()
+    >>> slice_of_anasig = proxy_anasig.load(time_slice=(1.*pq.s, 2.*pq.s))
+    >>> some_channel_of_anasig = proxy_anasig.load(channel_indexes=[0,5,10])
+
+    '''
+    _single_parent_objects = ('Segment', 'ChannelIndex')
+    _necessary_attrs = (('sampling_rate', pq.Quantity, 0),
+                                    ('t_start', pq.Quantity, 0))
+    _recommended_attrs = BaseNeo._recommended_attrs
+
+    def __init__(self, rawio=None, global_channel_indexes=None, block_index=0, seg_index=0):
+        self._rawio = rawio
+        self._block_index = block_index
+        self._seg_index = seg_index
+        if global_channel_indexes is None:
+            global_channel_indexes = slice(None)
+        total_nb_chan = self._rawio.header['signal_channels'].size
+        self._global_channel_indexes = np.arange(total_nb_chan)[global_channel_indexes]
+        self._nb_chan = self._global_channel_indexes.size
+
+        sig_chans = self._rawio.header['signal_channels'][self._global_channel_indexes]
+
+        assert np.unique(sig_chans['units']).size == 1, 'Channel do not have same units'
+        assert np.unique(sig_chans['dtype']).size == 1, 'Channel do not have same dtype'
+        assert np.unique(sig_chans['sampling_rate']).size == 1, \
+                    'Channel do not have same sampling_rate'
+
+        self.units = ensure_signal_units(sig_chans['units'][0])
+        self.dtype = sig_chans['dtype'][0]
+        self.sampling_rate = sig_chans['sampling_rate'][0] * pq.Hz
```

Comment:
The `sampling_period` attribute is missing here
```suggestion
        self.sampling_rate = sig_chans['sampling_rate'][0] * pq.Hz
        self.sampling_period = 1./self.sampling_rate
```